sunpy/map/mapbase.py

```diff
@@ -2560,7 +2608,8 @@ def _check_axes(self, axes, allow_non_wcsaxes=False, warn_different_wcs=False):
                                 "to this map when creating the axes.")
         elif warn_different_wcs and not axes.wcs.wcs.compare(self.wcs.wcs, tolerance=0.01):
             warnings.warn('The map world coordinate system (WCS) is different from the axes WCS. '
-                          'The map data axes may not correctly align with the coordinate axes.',
+                          'The map data axes may not correctly align with the coordinate axes. '
+                          'If this difference in WCS is intended, specify `different_wcs=True`.',
```

Comment:
```suggestion
                          'If this difference in WCS is intended, specify `autoalign=True`.',
```
Even with this fix I'm a bit confused by this sentence. Changing the message to `autoalign=False` in tandem with my comment above on setting `autoalign=None` by default might work.

Reply:
How about this:
```suggestion
                          'To automatically transform the data to the coordinate axes, specify `autoalign=True`.',
```

Reply:
That looks good to me. I thought the original message was saying "If you are knowingly using imshow to plot these two maps, even though they have different WCSes...", so I think your suggestion is much clearer.

Reply:
Accepted @Cadair's suggestion